docs/EVE-IMAGE-SOURCES.md

```diff
@@ -0,0 +1,616 @@
+# Source Tracing
+
+As an open source project, it is important for all of the sources for EVE to be open and available. While for some having this repository
+may suffice, EVE sets higher standards. Specifically, EVE desires that it be possible to trace from a given EVE image back to the
+sources of all software used in that binary compilation of EVE.
+
+This is done for several purposes:
+
+* Debugging, where knowledge of the precise source used is critical.
+* Auditing, where users auditing a particular distribution require confidence that they are reviewing precisely the correct source.
+* License compliance, where several licenses, notably the GPL family, require either distribution of source or links to source when distributing binary artifacts.
+
+This document will describe how to trace from a given compiled distribution of EVE all the way back to all sources.
+
+The stages are:
+
+1. Binary to source commit
+1. Binary to builder configuration
+1. Container images to container sources
+1. Container ources to contents:
+   * Local content
+   * Go modules
+   * External sources
+   * Alpine packages
+
+## Binary to source commit
+
+EVE is distributed in several formats.
+
+* Docker Images at [Docker Hub](https://hub.docker.com/r/lfedge/eve/). These are tagged with the version or commit of the image, as well as the architecture and hypervisor.
+* Release artifacts [on github](https://github.com/lf-edge/eve/releases). For example, [version 8.10.0](https://github.com/lf-edge/eve/releases/tag/8.10.0).
+
+If you are using a particular artifact, use the version provided to get your release version.
+
+* For Docker images with a [semver](https://semver.org/) tag, use the tag. For example, for `lfedge/eve:8.10.0-kvm`, use `8.10.0`.
+* For Docker images with a hash commit, use the commit. For example, for `lfedge/eve:0.0.0-master-5729285b-kvm-arm64`, use `5729285b`.
+* For release artifacts, use the version on the download page. FOr example, if you downloaded from https://github.com/lf-edge/eve/releases/download/8.10.0/amd64.rootfs.img, use `8.10.0`.
+
+If you have an image for which you do not have the hash or semver version, you should be able to retrieve the version from the image itself.
+
+* For Docker images, look in `/bits/eve_version`. For example:
+```
+$ docker run -it --entrypoint=sh lfedge/eve:0.0.0-master-0c6de671-kvm -c "cat /bits/eve_version"
```

Comment:
docker run lfedge/eve:0.0.0-master-0c6de671-kvm version
is less typing.

Reply:
I forgot that you can do that! Updating.